materialize-elasticsearch/transactor.go

```diff
@@ -187,52 +188,59 @@ func (t *transactor) Store(it *m.StoreIterator) (m.StartCommitFunc, error) {
 
 		doc := make(map[string]any)
 
-		for idx, v := range append(it.Key, it.Values...) {
-			if b, ok := v.([]byte); ok {
-				// An object or array field is received as raw JSON bytes. We currently only support
-				// objects.
-				v = json.RawMessage(b)
-			}
-			if s, ok := v.(string); b.floatFields[idx] && ok {
-				// ElasticSearch does not supporting indexing these special Float values.
-				if s == "Infinity" || s == "-Infinity" || s == "NaN" {
-					v = nil
-				}
-			}
-
-			doc[b.fields[idx]] = v
-		}
-		if b.docField != "" {
-			doc[b.docField] = it.RawJSON
-		}
-
 		var id string
 		if !b.deltaUpdates {
 			// Leave ID blank for delta updates so that ES will automatically generate one.
 			id = base64.RawStdEncoding.EncodeToString(it.PackedKey)
 		}
 
-		// The "create" action will fail if an item by the provided ID already exists, and "index"
-		// is like a PUT where it will create or replace. We could just use "index" all the time,
-		// but using "create" when we believe the item does not already exist provides a bit of
-		// extra consistency checking.
-		action := "create"
-		if it.Exists {
-			action = "index"
-		}
+		var action string
+		var bodyReader *bytes.Reader = bytes.NewReader([]byte{})
```

Comment:
Can we do `var bodyReader *bytes.Reader`? Or does it actually have to be non-nil when its use in the bulk indexer?

Reply:
@williamhbaker it needs to be non-nil in the case of deletions, I tried a nil first and the library panics